src/app/widgets/Visualizer/GCodeVisualizer.js

```diff
@@ -139,6 +139,10 @@ class GCodeVisualizer {
         this.frameIndex = frameIndex;
     }
 
+    getCurrentLocation() {
+        return this.vertices[this.frames[this.frameIndex].vertexIndex] ? this.vertices[this.frames[this.frameIndex].vertexIndex] : this.vertices[this.frames[this.frameIndex].vertexIndex - 1];
```

Comment:
I have this null check here because for some reason, the last frame would point to a `vertexIndex` just outside the range of the vertices array. Not sure why this happens. I can look into it further if you want.

Reply:
That is weird, you could look into it if you'd like but if this check doesn't cause any issues or side effects then it's fine to leave it alone for now.